src/utils.rs

```diff
@@ -0,0 +1,128 @@
+use std::collections::HashMap;
+use pkcs11_uri::{Pkcs11Uri};
+use crate::keywrap::pkcs11::Pkcs11KeyFileObject;
+
+extern crate serde_yaml;
+extern crate base64;
+
+// OAEPDefaultHash defines the default hash used for OAEP encryption; this
+// cannot be changed
+static OAEP_DEFAULT_HASH: &str = "sha1";
+
+
+
+// Pkcs11KeyFile describes the format of the pkcs11 (private) key file.
+// It also carries pkcs11 module related environment variables that are
+// transferred to the Pkcs11URI object and activated when the pkcs11 module is
+// used.
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+struct Pkcs11KeyFile {
+    uri: String,
+    env: HashMap<String, String>,
+}
+
+
+// Pkcs11Blob holds the encrypted blobs for all recipients; this is what we will put into the image's annotations
+#[derive(Serialize, Deserialize)]
+struct Pkcs11Blob {
+    version:    u32,              //`json:"version"`
+    recipients: Vec<Pkcs11Recipient>, // `json:"recipients"`
+}
+
+// Pkcs11Recipient holds the b64-encoded and encrypted blob for a particular
+// recipient
+#[derive(Serialize, Deserialize)]
+struct Pkcs11Recipient {
+    version: u32, //`json:"version"`
+    blob:    String, // `json:"blob"`
+    hash:    String, // `json:"hash,omitempty"`
+}
+
+
+pub fn parse_pkcs11_uri(uri: &str) -> Result<Pkcs11Uri, std::io::Error> {
+    Ok(Pkcs11Uri::try_from(uri).unwrap())
+}
+
+
+// ParsePkcs11KeyFile parses a pkcs11 key file holding a pkcs11 URI describing
+// a private key.
+// The file has the following yaml format:
+// pkcs11:
+//  - uri : <pkcs11 uri>                                                                  // An error is returned if the pkcs11 URI is malformed
+pub fn parse_pkcs11_key_file(yaml_bytes: &Vec<u8>)
+                             -> Result<Pkcs11KeyFileObject, std::io::Error> {
+    let s = serde_yaml::to_string(yaml_bytes).unwrap();
```

Comment:
Is it safe to blindly unwrap this? We will get a panic at runtime if the yaml parsing doesn't work out. Might be good to pass the error upwards with the ? syntax. That may require changing the result type, but it doesn't look like you're using the error in the result anyway.

Reply:
Thanks for the guidance on error handling. I've handled it in newer commits. I'm going to talk with Arron, too, to make sure we're on the same page about error handing.